### ISSUE TITLE
Deconvolution with known SNPs per sample - vcf_donor vs vcf_mixed

### ISSUE THREAD

**Issue by bednarsky** [theislab/hadge#43]
Hi once again hadge team!

I have used the pipeline to demultiplex multiple samples that were always contain cells from two donors, and it worked very well! Since I did not have a reference the cells were assigned to anonymous donors.

In the meanwhile, I received WGS data for all samples, and will soon have SNP calling results. Now, I would like to use this SNPs to assign specific donors to each cell. 

I am somewhat confused by the options to provide reference vcf files. From your documentation:
> Input data | Parameter
> -- | --
> Genotype reference (VCF) | params.vcf_donor
> Genotypes from referenced population (VCF) | params.vcf_mixed
where different genetic deconvolution methods are indicated further above to use different parameters, e.g., `freemuxlet` uses `Genotypes from referenced population (VCF)` only. 

So far, before I had the reference SNP data, I did not set any of these parameters, but only used the `common_variants_{method}` parameters in `nextflow.config`. 

My questions:
- Should I still set the `common_variants_{method}` parameters in `nextflow.config`?
- How should I set the params `vcf_donor` and `vcf_mixed`? Should I use `.vcf` files containing exactly the SNPs for the samples that were mixed and should be demultiplexed for both these parameters? 
- Should I still do variant calling? 
  - scsplit_variant: guess`False`, since input data will be gotten from `vcf_mixed`? 
  - vireo_variant: guess `True`, since I don't have SNPs per cell that would be given to the param `celldata`
- Am I correct to assume some methods don't assign specific donors (if they only use `vcf_mixed` or neither of the `vcf_*` params, like Souporcell)? 
  - If yes: Do these methods still provide SNPs that were used to classify into anonymous donors?
    - If yes: Are they automatically used by `hadge` to assign clear-label donors? 
      - If not autmatically assigned: Why not? Anything that speaks against doing that? Where can I find them? 

Thanks a lot for all your effort and expertise!

**Comment by wxicu** [comment]
Hi,

- In general, `vcf_donor` is the known SNPs per sample that you have.
- You can set `common_variants_{method}` to known SNPs per sample instead of common snps. This will speed up the process. But this is not necessary for all methods. I will explain it in details in the following points.
- Since you have already the donor reference, you can run demuxlet instead of freemuxlet, where vcf_donor will be used. No need to use common variants any more. But if you still want to run freemuxlet, just follow the previous point and set `common_variants_freemuxlet` to known SNPs per sample.
- For vireo, yes you still need to run cellsnp to get the SNPs per cell. Use known SNPs per cell for `common_variants_vireo` will reduce the run time. `vcf_donor` will also be used if specified.
- For Souporcell, you dont need to specify `common_variants_souporcell` since it has the option to read from `vcf_donor` (please set `params.use_known_genotype` to True). However, souporcell may raise error when there is a missing value for a sample's genotype somewhere. If so, you can use `common_variants_souporcell`.
- For scSplit, this is where `vcf_mixed` is used. `vcf_mixed` is created using freebayes to call all SNPs in the population. The result is per pool, not per sample in the pool. (So there is only one "sample" in the vcf file.) So you still need to call `scsplit_variant`. But you can use `common_variants_scsplit` to speed up the process. again, `vcf_donor` will also be used if specified.
- So in summary all tools can use vcf_donor if it is given. They can assign cells to specific donors, which means there is no anonymous donor ids any more, but the donor names given in the known snps per sample instead.

I hope i have explained clearly!

Best wishes,
Xichen

**Comment by bednarsky** [comment]
Amazing, thank you so much for this fast and elaborate answer!

**Comment by wxicu** [comment]
Please feel free to reopen this issue if you are still running into issues and we would be happy to help troubleshoot further.
